mkosi/__init__.py

```diff
@@ -610,7 +580,7 @@ def install_boot_loader(state: MkosiState) -> None:
         return
 
     with complete_step("Installing boot loader…"):
-        run(["bootctl", "install", "--root", state.root], env={"SYSTEMD_ESP_PATH": "/boot", **os.environ})
+        run(["bootctl", "install", "--root", state.root], env={"SYSTEMD_ESP_PATH": "/boot"})
```

Comment:
Probably also moot, since we don't guarantee upgradability, but this is where Debian loses that, since dpkg and its maintainer throw their hands into the air. I'm fine with continuing to have ESP on `/boot`, but BLS prefers `/efi`, so we could do that.

Reply:
yeah this will break badly, if /boot is fat32 stuff will just start exploding left and right

Reply:
I'm confused, what's the exact problem here? We install into /boot because we have to install somewhere, but afterwards this directory is just packed up into the EFI partition which systemd can mount wherever it wants when booting.

Reply:
as long as this doesn't mean that /boot ends up as fat32 it's fine - we should have the /efi mountpoint in the image for read-only cases, is that there? any reason not to just install to /efi directly?

Reply:
> as long as this doesn't mean that /boot ends up as fat32 it's fine - we should have the /efi mountpoint in the image for read-only cases, is that there? any reason not to just install to /efi directly?

The dracut kernel-install script and other things hardcode /boot as the output directory.

So we just need to make sure that the EFI partition ends up being mounted at /efi in the booted image?

Reply:
> The dracut kernel-install script and other things hardcode /boot as the output directory.
Just an late FYI 
The reason why dracut ( and presumably other things ) have /boot hardcoded is because GRUB 2 bootloader expects the initrd to be installed in /boot which may or may not affect your change here.